bee-network/bee-autopeering/src/discovery/messages.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{local::services::ServiceMap, peer::Peer, proto, request::Request};
+
+use prost::{bytes::BytesMut, DecodeError, EncodeError, Message as _};
+
+use std::{
+    fmt,
+    net::{AddrParseError, IpAddr, SocketAddr},
+};
+
+#[derive(Debug, thiserror::Error)]
+pub(crate) enum Error {
+    #[error("The peer did not announce any services.")]
+    NoServices,
+    #[error("Invalid source ip address. Cause: {0}.")]
+    InvalidSourceIpAddress(AddrParseError),
+    #[error("Invalid target ip address. Cause: {0}.")]
+    InvalidTargetIpAddress(AddrParseError),
+    #[error("{0}")]
+    ProtobufDecode(#[from] DecodeError),
+    #[error("{0}")]
+    ProtobufEncode(#[from] EncodeError),
+}
+
+#[derive(Clone, Copy)]
+pub(crate) struct VerificationRequest {
+    version: u32,
+    network_id: u32,
+    timestamp: u64,
+    source_addr: SocketAddr,
+    target_addr: IpAddr,
+}
+
+impl VerificationRequest {
+    pub fn new(version: u32, network_id: u32, source_addr: SocketAddr, target_addr: IpAddr) -> Self {
+        let timestamp = crate::time::unix_now_secs();
+
+        Self {
+            version,
+            network_id,
+            timestamp,
+            source_addr,
+            target_addr,
+        }
+    }
+
+    pub fn version(&self) -> u32 {
+        self.version
+    }
+
+    pub fn network_id(&self) -> u32 {
+        self.network_id
+    }
+
+    pub fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+
+    pub fn source_addr(&self) -> SocketAddr {
+        self.source_addr
+    }
+
+    pub fn from_protobuf(bytes: &[u8]) -> Result<Self, Error> {
+        let proto::Ping {
```

Comment:
Just curious, why don't we use the same name, ping ?

Reply:
Because I think:
(a) it would not be a consistent: All other message types follow the Request/Response naming convention:
```rust
pub(crate) enum MessageType {
    VerificationRequest = DISCOVERY_MSG_TYPE_MIN,
    VerificationResponse,
    DiscoveryRequest,
    DiscoveryResponse,
    PeeringRequest = PEERING_MSG_TYPE_MIN,
    PeeringResponse,
    DropRequest,
}
```
(b) it is more descriptive
(c) could be confused with Ping/Pong message spec from the IETF (https://datatracker.ietf.org/doc/html/rfc6455#section-5.5.2)
(d) even in the Hornet/Hive code you find the term verification request somewhere (I think in a code comment).